source/binding/Bitstream.cpp

```diff
@@ -178,15 +202,33 @@ static size_t bitstreamCastRemainingSize(const T& destination, size_t srcSize) {
     // Calculate remaining size to dynamically fill.
     auto [destFillMultiplier, destFillFixedSize] =
         dynamicBitstreamSize(destination, BitstreamSizeMode::DestFill);
-    if (srcSize < destFillFixedSize || (srcSize - destFillFixedSize) % destFillMultiplier != 0)
+    if (srcSize < destFillFixedSize || (srcSize - destFillFixedSize) % destFillMultiplier != 0) {
+        if (destEmptyMultiplier > 0 && (srcSize - destEmptyFixedSize) % destEmptyMultiplier == 0)
+            return 0; // only for "with" range
         return srcSize + 1;
+    }
 
     // Size to fill the first dynamically sized item.
     return srcSize - destFillFixedSize;
 }
 
+/// Formats the width of a dynamically-sized bitstream to a formula like 8×n+3 in an error message
+template<typename T>
+static std::string formatWidth(const T& bitstream, bool destination = false) {
```

Comment:
It doesn't seem like this parameter should be defaulted. Also, would it make the call sites more clear if you pass the BitstreamSizeMode instead?

Reply:
Yes, I replace "bool destination = false" with "BitstreamSizeMode mode".